File: SofiForce.Web/ClientApp/src/app/core/Models/EntityModels/ClientSurveyModel.ts
import { SurveyModel } from "./SurveyModel";

export interface ClientSurveyModel {
    clientServeyId: number;
    surveyId: number;
    surveyCode: string;

    clientId: number | null;
    clientCode: string;

    branchId: number | null;
    branchCode: string;

    representativeId: number | null;
    representativeCode: string;

    isClosed: boolean;


    serveyStatusId: number | null;
    createDate: Date | null;
    createTime: Date | null;
    startDate: Date | null;
    startTime: Date | null;

    notes:string;

    inZone: boolean ;
    distance: number ;
    latitude: number | null;
    longitude: number | null;


    serveyModel:SurveyModel;
}

File: SofiForce.Models/Models/EntityModels/ClientSurveyModel.cs
using System;
using System.Collections.Generic;

namespace Models
{
    public class ClientSurveyModel
    {

        public string Id { get; set; }
        public bool? IsSynced { get; set; } = false;
        public DateTime? SyncDate { get; set; }

        public double? ClientServeyId { get; set; }
        public int SurveyId { get; set; }
        public int? ClientId { get; set; }
        public int? BranchId { get; set; }

        public int? RepresentativeId { get; set; }
        public int? ServeyStatusId { get; set; }
        public DateTime? CreateDate { get; set; }
        public DateTime? CreateTime { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? StartTime { get; set; }


        public bool? InZone { get; set; }
        public double? Distance { get; set; }

        public double? Latitude { get; set; }
        public double? Longitude { get; set; }

        public string Notes { get; set; }
        public SurveyModel ServeyModel { get; set; } = new SurveyModel();
    }
}
